src/async_context_frame.cc

```diff
@@ -0,0 +1,89 @@
+#include "async_context_frame.h"  // NOLINT(build/include_inline)
+
+#include "env-inl.h"
+#include "node_errors.h"
+#include "node_external_reference.h"
+#include "tracing/traced_value.h"
+#include "util-inl.h"
+
+#include "debug_utils-inl.h"
+
+#include "v8.h"
+
+using v8::Context;
+using v8::Isolate;
+using v8::Local;
+using v8::Object;
+using v8::String;
+using v8::Value;
+
+namespace node {
+namespace async_context_frame {
+
+//
+// Scope helper
+//
+Scope::Scope(Isolate* isolate, Local<Value> object) : isolate_(isolate) {
+  auto prior = exchange(isolate, object);
+  prior_.Reset(isolate, prior);
+}
+
+Scope::~Scope() {
+  auto value = prior_.Get(isolate_);
+  set(isolate_, value);
+}
+
+Local<Value> current(Isolate* isolate) {
+  return isolate->GetContinuationPreservedEmbedderData();
```

Comment:
```suggestion
  auto env = Environment::GetCurrent(isolate);
  if (!env->options()->async_context_frame) {
    return;
  }
  return isolate->GetContinuationPreservedEmbedderData();
```

At least I assume we should add it similar as in `set`.

Reply:
That was intentional. Set mutates, Get does not. The option check is to protect CPED from being modified by accident.